src/components/carousel/index.tsx

```diff
@@ -14,6 +15,7 @@ type SharedProps = {
   startIndex?: number;
   onSlideClick?: (index: number) => void;
   onSlideSelect?: (index: number) => void;
+  withCustomPagination?: boolean;
```

Comment:
for me this is a bit confusing because we already have a pagination. How about doing something like:
```suggestion
  paginationType?: "numbers" | "images";
```

Reply:
I wouldn't split them by type, because we enable thumbnail pagination on fullscreen mode, without a specific prop, so I changed it to `withNumbersPagination` to make it cleaner.